tests/lib/uuid/src/main.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) 2024, SECO Mind Srl
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/ztest.h>
+#include <zephyr/sys/uuid.h>
+
+ZTEST(uuid, test_uuid_v4)
+{
+	uuid_t uuid_1 = {0};
+	uuid_t uuid_2 = {0};
+
+	uuid_generate_v4(uuid_1);
+	/* Check version and variant fields */
+	zassert_equal(uuid_1[6U] >> 4U, 4U, "Generated UUID v4 contains an incorrect 'ver' field");
+	zassert_equal(uuid_1[8U] >> 6U, 2U, "Generated UUID v4 contains an incorrect 'var' field");
+	/* Check another generated UUID is different */
+	uuid_generate_v4(uuid_2);
+	zassert(memcmp(uuid_1, uuid_2, UUID_SIZE) != 0, "The two generated UUID v4 are equal.");
+}
+
+ZTEST(uuid, test_uuid_v5)
+{
+	uuid_t namespace;
+	int result;
+	uuid_t uuid;
+	char uuid_str[UUID_STR_LEN + 1];
+
+	result = uuid_from_string("6ba7b810-9dad-11d1-80b4-00c04fd430c8", namespace);
+	zassert_true(result == 0, "uuid_from_string returned an error");
+	result = uuid_generate_v5(namespace, "www.example.com", 15, uuid);
+	zassert_true(result == 0, "uuid_generate_v5 returned an error");
+	result = uuid_to_string(uuid, uuid_str, UUID_STR_LEN + 1);
+	zassert_true(result == 0, "uuid_from_string returned an error");
+	zassert_true(strcmp("2ed6657d-e927-568b-95e1-2665a8aea6a2", uuid_str) == 0,
+		     "uuid_str != 2ed6657d-e927-568b-95e1-2665a8aea6a2");
+}
+
+ZTEST(uuid, test_uuid_from_string)
+{
+	const char *first_uuid_v4_string = "44b35f73-cfbd-43b4-8fef-ca7baea1375f";
+	const char *second_uuid_v4_string = "6f2fd4cb-94a0-41c7-8d27-864c6b13b8c0";
+	const char *third_uuid_v4_string = "8f65dbbc-5868-4015-8523-891cc0bffa58";
+	const char *first_uuid_v5_string = "0575a569-51eb-575c-afe4-ce7fc03bcdc5";
+
+	uuid_t first_uuid_v4;
+	uuid_t second_uuid_v4;
+	uuid_t third_uuid_v4;
+	uuid_t first_uuid_v5;
+
+	uint8_t expected_first_uuid_v4_byte_array[16u] = {0x44, 0xb3, 0x5f, 0x73, 0xcf, 0xbd,
```

Comment:
More drive-by: this is clearly byte-packing a **big endian** UUID.  The first element is a 32 bit dword and should be stored "73, 5f, b3, 44" on little endian systems.